### PR TITLE
Make apply function public

### PR DESCRIPTION
**Pull request by shixuan-fan**
Address #9293

### CODE REVIEW

File: presto-docs/src/main/sphinx/functions/lambda.rst [hunk 1]
```diff
@@ -24,6 +24,13 @@ Most SQL expressions can be used in a lambda body, with a few exceptions:
 Lambda Functions
 ----------------
 
+.. function:: apply(T, function<T,U>) -> U
+
+    Apply the lambda function on the given variable::
```
Comment (wenleix) [inline presto-docs/src/main/sphinx/functions/lambda.rst]: Does `... on the giving value` sounds better? @haozhun

File: presto-docs/src/main/sphinx/functions/lambda.rst [hunk 2]
```diff
@@ -24,6 +24,13 @@ Most SQL expressions can be used in a lambda body, with a few exceptions:
...
+
+        SELECT apply(5, x -> x + 1); -- 6
+        SELECT apply(1, x -> apply(x + 7, y -> y * 5) + 1); -- 41
```
Comment (wenleix) [inline presto-docs/src/main/sphinx/functions/lambda.rst]: Let's use an example demonstrate the application of apply in using some input multiple times, e.g.

```
SELECT apply(5, IF(x % 2 == 0, x / 2, x * 3 + 1); -- 16
SELECT apply(16, IF(x % 2 == 0, x / 2, x * 3 + 1)); -- 8
```

File: presto-docs/src/main/sphinx/functions/lambda.rst (hunk 1 above)
Comment (shixuan-fan) [reply presto-docs/src/main/sphinx/functions/lambda.rst]: `value` sounds like it will only accept a constant value. Maybe we could use `to the first argument`?

File: presto-docs/src/main/sphinx/functions/lambda.rst (hunk 2 above)
Comment (shixuan-fan) [reply presto-docs/src/main/sphinx/functions/lambda.rst]: Isn't this more like an example of `IF`? :-|

File: presto-docs/src/main/sphinx/functions/lambda.rst (hunk 1 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/functions/lambda.rst]: Let’s say “Apply the function to the given value”

File: presto-docs/src/main/sphinx/functions/lambda.rst (hunk 2 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/functions/lambda.rst]: Something involving substr and length is a good example, where you need the value and the length of the value (for the offset or length calculation).

File: presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java [hunk 3]
```diff
@@ -573,7 +574,8 @@ public FunctionRegistry(TypeManager typeManager, BlockEncodingSerde blockEncodin
                 .function(DECIMAL_MOD_FUNCTION)
                 .functions(ARRAY_TRANSFORM_FUNCTION, ARRAY_REDUCE_FUNCTION)
                 .functions(MAP_FILTER_FUNCTION, MAP_TRANSFORM_KEY_FUNCTION, MAP_TRANSFORM_VALUE_FUNCTION)
-                .function(TRY_CAST);
+                .function(TRY_CAST)
+                .function(APPLY_FUNCTION);
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java]: Leave try_cast last so that we don’t have to change an unrelated line

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/ApplyFunction.java [hunk 4]
```diff
@@ -59,7 +59,7 @@ private ApplyFunction()
     @Override
     public boolean isHidden()
     {
-        return true;
+        return false;
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/ApplyFunction.java]: Update description here as well

**Comment by shixuan-fan** [comment]
Close this because it is not required any more.
